manticore/platforms/linux.py

```diff
@@ -1296,14 +1311,12 @@ def sys_open(self, buf, flags, mode):
                     filename = os.path.abspath(self.program)
                 else:
                     logger.info("FIXME!")
-            mode = {os.O_RDWR: 'r+', os.O_RDONLY: 'r', os.O_WRONLY: 'w'}[flags&7]
-
-            f = self._sys_open_get_file(filename, flags, mode)
-            logger.debug("Opening file %s for %s real fd %d",
-                         filename, mode, f.fileno())
-        # FIXME(theo) generic exception
-        except Exception as e:
-            logger.info("Could not open file %s. Reason %s" % (filename, str(e)))
+
+            f = self._sys_open_get_file(filename, flags)
+            logger.debug("Opening file %s for real fd %d",
+                         filename, f.fileno())
+        except IOError as e:
+            logger.info("Could not open file %s. Reason: %s" % (filename, str(e)))
```

Comment:
Just saw this..

Is better not to use the % explicitly there.
logger.info("Could not open file %s. Reason: %s", filename, str(e)) 
This will not format the string if not  log is not active.